packages/ppx_deriving_madcast/ppx_deriving_madcast.0.2/opam

```diff
@@ -0,0 +1,50 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "Library deriving cast functions based on their types"
+description: """
+This package provides a PPX that allows to derive cast functions based
+on their types.
+
+For instance, `[%madcast: ('a * string) -> ('a * int)]` would be replaced by:
+
+    fun (x, y) ->
+      (x,
+       try int_of_string y
+       with Failure _ -> failwith "madcast: string -> int")
+"""
+maintainer: ["Niols <niols@niols.fr>"]
+authors: ["Kerl <kerl@wkerl.me>" "Niols <niols@niols.fr>"]
+license: "LGPL-3.0-or-later"
+tags: ["syntax"]
+homepage: "https://github.com/LesBoloss-es/ppx_deriving_madcast"
+bug-reports: "https://github.com/LesBoloss-es/ppx_deriving_madcast/issues"
+depends: [
+  "dune" {>= "2.5"}
+  "ocaml"
+  "odoc" {with-test}
+  "ppx_deriving" {>= "5.0"}
+  "ppxlib"
```

Comment:
There are lower-bound errors which seem linked to ppxlib. You probably need to add a lower-bound constraint to this dependency.

Reply:
Lower-bound on ppxlib I think. This will probably end up indirectly limiting the range of ocaml versions that are supported.

Not sure what the proper lower-bound should be set to. I'll rerun the failed jobs so we can get the logs (I think they have expired) and you can get more info about the failure.

Reply:
The error log:

```

#=== ERROR while compiling ppx_deriving_madcast.0.2 ===========================#
# context              2.2.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.4.11.2 | pinned(https://github.com/LesBoloss-es/ppx_deriving_madcast/archive/v0.2.tar.gz)
# path                 ~/.opam/4.11/.opam-switch/build/ppx_deriving_madcast.0.2
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p ppx_deriving_madcast -j 31 @install
# exit-code            1
# env-file             ~/.opam/log/ppx_deriving_madcast-7-c816fe.env
# output-file          ~/.opam/log/ppx_deriving_madcast-7-c816fe.out
### output ###
# File "/home/opam/.opam/4.11/lib/ppx_derivers/ppx_derivers.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.11/lib/ocaml-compiler-libs/bytecomp/ocaml-compiler-libs.bytecomp.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.11/lib/ocaml-compiler-libs/common/ocaml-compiler-libs.common.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.11/lib/ocaml-compiler-libs/shadow/ocaml-compiler-libs.shadow.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.11/lib/ocaml-compiler-libs/toplevel/ocaml-compiler-libs.toplevel.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
#       ocamlc src/.madcast.objs/byte/madcast.{cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.11/bin/ocamlc.opt -w -40 -g -bin-annot -I src/.madcast.objs/byte -I /home/opam/.opam/4.11/lib/base -I /home/opam/.opam/4.11/lib/base/base_internalhash_types -I /home/opam/.opam/4.11/lib/base/caml -I /home/opam/.opam/4.11/lib/base/shadow_stdlib -I /home/opam/.opam/4.11/lib/ocaml-compiler-libs/common -I /home/opam/.opam/4.11/lib/ocaml-compiler-libs/shadow -I /home/opam/.opam/4.11/lib/ocaml-migrate-parsetree -I /home/opam/.opam/4.11/lib/ocaml/compiler-libs -I /home/opam/.opam/4.11/lib/ppx_derivers -I /home/opam/.opam/4.11/lib/ppx_deriving/api -I /home/opam/.opam/4.11/lib/ppxlib -I /home/opam/.opam/4.11/lib/ppxlib/ast -I /home/opam/.opam/4.11/lib/ppxlib/print_diff -I /home/opam/.opam/4.11/lib/ppxlib/traverse_builtins -I /home/opam/.opam/4.11/lib/result -I /home/opam/.opam/4.11/lib/sexplib0 -I /home/opam/.opam/4.11/lib/stdio -intf-suffix .ml -no-alias-deps -open Madcast__ -o src/.madcast.objs/byte/madcast.cmo -c -impl src/madcast.pp.ml)
# File "src/madcast.ml", line 109, characters 26-28:
# 109 |           assert (casts = []);
#                                 ^^
# Error: This variant expression is expected to have type Ppxlib.Longident.t
#        The constructor [] does not belong to type Ppxlib.Longident.t
#     ocamlopt src/.madcast.objs/native/madcast.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.11/bin/ocamlopt.opt -w -40 -g -I src/.madcast.objs/byte -I src/.madcast.objs/native -I /home/opam/.opam/4.11/lib/base -I /home/opam/.opam/4.11/lib/base/base_internalhash_types -I /home/opam/.opam/4.11/lib/base/caml -I /home/opam/.opam/4.11/lib/base/shadow_stdlib -I /home/opam/.opam/4.11/lib/ocaml-compiler-libs/common -I /home/opam/.opam/4.11/lib/ocaml-compiler-libs/shadow -I /home/opam/.opam/4.11/lib/ocaml-migrate-parsetree -I /home/opam/.opam/4.11/lib/ocaml/compiler-libs -I /home/opam/.opam/4.11/lib/ppx_derivers -I /home/opam/.opam/4.11/lib/ppx_deriving/api -I /home/opam/.opam/4.11/lib/ppxlib -I /home/opam/.opam/4.11/lib/ppxlib/ast -I /home/opam/.opam/4.11/lib/ppxlib/print_diff -I /home/opam/.opam/4.11/lib/ppxlib/traverse_builtins -I /home/opam/.opam/4.11/lib/result -I /home/opam/.opam/4.11/lib/sexplib0 -I /home/opam/.opam/4.11/lib/stdio -intf-suffix .ml -no-alias-deps -open Madcast__ -o src/.madcast.objs/native/madcast.cmx -c -impl src/madcast.pp.ml)
# File "src/madcast.ml", line 109, characters 26-28:
# 109 |           assert (casts = []);
#                                 ^^
# Error: This variant expression is expected to have type Ppxlib.Longident.t
#        The constructor [] does not belong to type Ppxlib.Longident.t
```

Hope this helps you find the correct version that you need to set as a lowerbound for ppxlib.

Reply:
I manage to reproduce the error locally with `ppxlib` from `0.9.0` to `0.14.0`, so let's try with a `>= 0.15.0` constraint. cf 1b5ff5a.